To: vim-dev@vim.org
Subject: Patch 7.3.017
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.017
Problem:    smatch reports errors.
Solution:   Fix the reported errors. (Dominique Pelle)
Files:	    src/spell.c, src/syntax.c


*** ../vim-7.3.016/src/spell.c	2010-09-18 13:36:41.000000000 +0200
--- src/spell.c	2010-09-29 18:25:09.000000000 +0200
***************
*** 4951,4957 ****
      char_u	*si_info;	/* info text chars or NULL  */
      int		si_region_count; /* number of regions supported (1 when there
  				    are no regions) */
!     char_u	si_region_name[16]; /* region names; used only if
  				     * si_region_count > 1) */
  
      garray_T	si_rep;		/* list of fromto_T entries from REP lines */
--- 4951,4957 ----
      char_u	*si_info;	/* info text chars or NULL  */
      int		si_region_count; /* number of regions supported (1 when there
  				    are no regions) */
!     char_u	si_region_name[17]; /* region names; used only if
  				     * si_region_count > 1) */
  
      garray_T	si_rep;		/* list of fromto_T entries from REP lines */
***************
*** 9530,9536 ****
  	    if (aspath)
  		/* Use directory of an entry with path, e.g., for
  		 * "/dir/lg.utf-8.spl" use "/dir". */
! 		vim_strncpy(buf, curbuf->b_s.b_p_spl, lstart - curbuf->b_s.b_p_spl - 1);
  	    else
  		/* Copy the path from 'runtimepath' to buf[]. */
  		copy_option_part(&rtp, buf, MAXPATHL, ",");
--- 9530,9537 ----
  	    if (aspath)
  		/* Use directory of an entry with path, e.g., for
  		 * "/dir/lg.utf-8.spl" use "/dir". */
! 		vim_strncpy(buf, curbuf->b_s.b_p_spl,
! 					    lstart - curbuf->b_s.b_p_spl - 1);
  	    else
  		/* Copy the path from 'runtimepath' to buf[]. */
  		copy_option_part(&rtp, buf, MAXPATHL, ",");
***************
*** 9539,9551 ****
  		/* Use the first language name from 'spelllang' and the
  		 * encoding used in the first loaded .spl file. */
  		if (aspath)
! 		    vim_strncpy(buf, curbuf->b_s.b_p_spl, lend - curbuf->b_s.b_p_spl);
  		else
  		{
  		    /* Create the "spell" directory if it doesn't exist yet. */
  		    l = (int)STRLEN(buf);
  		    vim_snprintf((char *)buf + l, MAXPATHL - l, "/spell");
! 		    if (!filewritable(buf) != 2)
  			vim_mkdir(buf, 0755);
  
  		    l = (int)STRLEN(buf);
--- 9540,9553 ----
  		/* Use the first language name from 'spelllang' and the
  		 * encoding used in the first loaded .spl file. */
  		if (aspath)
! 		    vim_strncpy(buf, curbuf->b_s.b_p_spl,
! 						  lend - curbuf->b_s.b_p_spl);
  		else
  		{
  		    /* Create the "spell" directory if it doesn't exist yet. */
  		    l = (int)STRLEN(buf);
  		    vim_snprintf((char *)buf + l, MAXPATHL - l, "/spell");
! 		    if (filewritable(buf) != 2)
  			vim_mkdir(buf, 0755);
  
  		    l = (int)STRLEN(buf);
*** ../vim-7.3.016/src/syntax.c	2010-08-15 21:57:32.000000000 +0200
--- src/syntax.c	2010-09-29 18:28:26.000000000 +0200
***************
*** 7600,7609 ****
  		/*
  		 * Copy characters from arg[] to buf[], translating <> codes.
  		 */
! 		for (p = arg, off = 0; off < 100 && *p; )
  		{
  		    len = trans_special(&p, buf + off, FALSE);
! 		    if (len)		    /* recognized special char */
  			off += len;
  		    else		    /* copy as normal char */
  			buf[off++] = *p++;
--- 7600,7609 ----
  		/*
  		 * Copy characters from arg[] to buf[], translating <> codes.
  		 */
! 		for (p = arg, off = 0; off < 100 - 6 && *p; )
  		{
  		    len = trans_special(&p, buf + off, FALSE);
! 		    if (len > 0)	    /* recognized special char */
  			off += len;
  		    else		    /* copy as normal char */
  			buf[off++] = *p++;
*** ../vim-7.3.016/src/version.c	2010-09-29 17:26:57.000000000 +0200
--- src/version.c	2010-09-29 17:47:27.000000000 +0200
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     17,
  /**/

-- 
A computer without Windows is like a fish without a bicycle.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\        download, build and distribute -- http://www.A-A-P.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
